File: 0643.MaximumAverageSubarrayI/findMaxAverage.ts
export default function findMaxAverage(nums: number[], k: number): number {
  let sum = 0;

  for (let i = 0; i < k; i++) {
    sum += nums[i];
  }

  let max = sum;

  for (let i = k; i < nums.length; i++) {
    sum += nums[i] - nums[i - k];

    max = Math.max(max, sum);
  }

  return max / k;
}

File: 0643.MaximumAverageSubarrayI/findMaxAverage.go
package solution

func findMaxAverage(nums []int, k int) float64 {
	sum := 0

	for i := 0; i < k; i++ {
		sum += nums[i]
	}

	max := sum

	for i := k; i < len(nums); i++ {
		sum += nums[i] - nums[i-k]

		if sum > max {
			max = sum
		}
	}

	return float64(max) / float64(k)
}
